### PR TITLE
Extend tsconfigs from root

### LINKED ISSUE #116: Add root tsconfig file
TODO
- Add `tsconfig.json` file to root of project
- Where possible extend from this root tsconfig file instead of duplicating the configuration

### PR DESCRIPTION
**Pull request by mpetrun5**
Closes: #116

### CODE REVIEW

File: packages/cli/tsconfig.json [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
-    "declarationDir": "build/types",
-    "target": "es5",
-    "module": "commonjs",
-    "moduleResolution": "node",
-    "declaration": true,
-    "sourceMap": true,
+    "lib": [
+      "es2015",
+      "es5",
+      "dom",
```
Comment (mpetrunic) [inline packages/cli/tsconfig.json:8]: A bit odd that we use dom globals in cli though.
Also we can probably add newer es libs

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    "noImplicitReturns": true,
+    "noUnusedLocals": true,
+    "module": "commonjs",
+    "sourceMap": true,
+    "target": "es5",
+    "resolveJsonModule": true,
+    "strictNullChecks": true
+  },
+  "include": [
+    "./src/**/*"
```
Comment (mpetrunic) [inline tsconfig.json:26]:
```suggestion
    "./src/**/*.ts"
```

**Review by mpetrunic — COMMENTED** [review]
Cehck if everything is builded as expected

File: packages/cli/tsconfig.json (hunk 1 above)
Comment (mpetrun5) [reply packages/cli/tsconfig.json:8]: Use it for WebAssembly and URL
Removed in other places where it is not used

**Review by MakMuftic — DISMISSED** [review]
LGTM

**Review by dOrgJelli — APPROVED** [review]

**Pull request merged**